mypy/meet.py

```diff
@@ -451,6 +460,24 @@ def _type_object_overlap(left: Type, right: Type) -> bool:
     elif isinstance(right, CallableType):
         right = right.fallback
 
+    if isinstance(left, Parameters):
+        if not isinstance(right, Parameters):
```

Comment:
I think that we can consider all pairs of parameters overlapping (left and right both instances of `Parameters`), which will simplify the logic a bit. See my other comment.

Reply:
Okay great, that'll simplify things a lot!